rpc_proxy.go

```diff
@@ -279,17 +298,27 @@ func (p *rpcProxy) isHandling(resp http.ResponseWriter,
 	if p.grpcWebProxy.IsGrpcWebRequest(req) ||
 		p.grpcWebProxy.IsGrpcWebSocketRequest(req) {
 
-		log.Infof("Handling gRPC web request: %s", req.URL.Path)
-		p.grpcWebProxy.ServeHTTP(resp, req)
+		if !p.hasStarted() {
+			resp.WriteHeader(http.StatusServiceUnavailable)
+			_, _ = resp.Write(make([]byte, 0))
+		} else {
+			log.Infof("Handling gRPC web request: %s", req.URL.Path)
+			p.grpcWebProxy.ServeHTTP(resp, req)
+		}
 
 		return true
 	}
 
 	// Normal gRPC requests are also easy to identify. These we can
 	// send directly to the lnd proxy's gRPC server.
 	if isGrpcRequest(req) {
-		log.Infof("Handling gRPC request: %s", req.URL.Path)
-		p.grpcServer.ServeHTTP(resp, req)
+		if !p.hasStarted() {
+			resp.WriteHeader(http.StatusServiceUnavailable)
+			_, _ = resp.Write(make([]byte, 0))
```

Comment:
Comment above may moreso apply here. Another route would be an http middleware chain (gRPC package) that checked this field and returned a response if things weren't fully active yet.